LayoutTests/http/wpt/mediasession/toggleCapture.html

```diff
@@ -0,0 +1,146 @@
+        frame2.contentWindow.startCapture(true, false)
+    ]);
+
+    let resultsPromise = Promise.all([
+        frame1.contentWindow.waitForToggleAction("togglecamera"),
+        frame1.contentWindow.waitForToggleAction("togglemicrophone"),
+        frame2.contentWindow.waitForNoToggleAction("togglecamera"),
+        frame2.contentWindow.waitForToggleAction("togglemicrophone"),
+    ]);
+    internals.setPageMuted("capturedevices");
+    assert_array_equals(await resultsPromise, [false, false, null, false]);
+
+    assert_equals(actionCount, 0);
+
+    resultsPromise = Promise.all([
+        frame1.contentWindow.waitForToggleAction("togglecamera"),
+        frame1.contentWindow.waitForToggleAction("togglemicrophone"),
+        frame2.contentWindow.waitForNoToggleAction("togglecamera"),
+        frame2.contentWindow.waitForToggleAction("togglemicrophone"),
+    ]);
+    internals.setPageMuted("");
+    assert_array_equals(await resultsPromise, [true, true, null, true]);
+
+    assert_equals(actionCount, 0);
+
+   frame1.contentWindow.stopCapture();
+   frame2.contentWindow.stopCapture();
+}, "togglecamera and togglemicrophone on various iframes");
+
+async function testActionEventOrder(track, action)
+{
+    let isActionHandled = false;
+    let isMuteEventFired = false;
+    let promiseAction = new Promise((resolve, reject) => {
+        setTimeout(() => reject("promiseAction timed out"), 5000);
+        navigator.mediaSession.setActionHandler(action, details => {
+            isActionHandled = true;
+            assert_false(isMuteEventFired);
+            resolve(details.isActivating);
+        });
+    });
+
+    let promiseMuteEvent = new Promise((resolve, reject) => {
+        setTimeout(() => reject("promiseMuteEvent mute timed out"), 5000);
+        track.onmute = () => {
+            isMuteEventFired = true;
+            assert_true(isActionHandled);
+            resolve();
+        };
+    });
+
+    internals.setPageMuted(action === "togglescreenshare" ? "screencapture" : "capturedevices");
+    await Promise.all([promiseAction, promiseMuteEvent]);
+
+    isActionHandled = false;
+    isMuteEventFired = false;
+    promiseAction = new Promise((resolve, reject) => {
+        setTimeout(() => reject("promiseAction timed out"), 5000);
+        navigator.mediaSession.setActionHandler(action, details => {
+            isActionHandled = true;
+            assert_false(isMuteEventFired);
+            resolve(details.isActivating);
+        });
+    });
+
+    promiseMuteEvent = new Promise((resolve, reject) => {
+        setTimeout(() => reject("promiseMuteEvent unmmute timed out"), 5000);
+        track.onunmute = () => {
+            isMuteEventFired = true;
+            assert_true(isActionHandled);
+            resolve();
+        };
+    });
+
+    internals.setPageMuted("");
+    await Promise.all([promiseAction, promiseMuteEvent]);
+}
+
+promise_test(async (test) => {
+    if (!window.internals)
```

Comment:
would be nicer to not test for window.internals here, would make migration to a UA-agnostic test easier

Reply:
Done